lisp/ess-r-xref.el

```diff
@@ -65,6 +74,27 @@
       (goto-char (match-beginning 0))
       (read (current-buffer)))))
 
+(defun ess-r-xref--pkg-srcfile (symbol r-src-file)
+  "Look in the source directory of the R package containing symbol SYMBOL for R-SRC-FILE."
+  (let* ((env-name (ess-string-command (format ".ess_pkg_name(\"%s\")\n" symbol)))
+         (pkg (if (string-equal env-name "R_GlobalEnv")
+                  ;; This probably shouldn't happen, but just in case...
+                  (error "Can't find package for symbol %s." symbol)
+                env-name))
+         (dir (or (assoc-default pkg ess-r-xref-pkg-sources)
+                  (when ess-r-xref-pkg-source-dir
+                    (expand-file-name pkg ess-r-xref-pkg-source-dir))
+                  (read-directory-name
+                   (format "%s package source dir: " pkg) nil nil t)))
```

Comment:
Let's not do this for now. It would mean that this backend is the dead end for R sources. I do like the dump idea, which should be the ending backend. I have looked into it and will implement it once #479 has been merged.

Reply:
How about hiding it behind a `yes-or-no-p` prompt? I do think it makes sense to allow a user to interactively tell ESS about the location of source packages.

Reply:
It would not solve the issue, y-or-n before dump backend is still too intrusive. Besides, I am not sure this feature is as useful as it sounds. We used to ask for file when debugger location couldn't be found, but it wasn't that useful and quite annoying.

Such options also encourage lazy behavior. People will be navigating to package locations (thus wasting time) instead of customizing one variable and be done with it.